adapters/repos/db/vector/hnsw/delete.go

```diff
@@ -343,14 +344,14 @@ func (h *hnsw) isOnlyNode(needle *vertex, denyList helpers.AllowList) bool {
 	return true
 }
 
-func (h *hnsw) hasTombstone(id int) bool {
+func (h *hnsw) hasTombstone(id int64) bool {
 	h.RLock()
 	defer h.RUnlock()
 	_, ok := h.tombstones[id]
 	return ok
 }
 
-func (h *hnsw) addTombstone(id int) {
+func (h *hnsw) addTombstone(id int64) {
 	h.Lock()
 	h.tombstones[id] = struct{}{}
 	h.Unlock()
```

Comment:
see above, all node ids should be represented by `uint64`, removing the need for the conversions.